remove duplicate tabular objects created when latexing summary_col

natashawatkins opened the pull request:
Best solution I've thought of to #3701 

- uses a regex to find where duplicate tabular objects are created from summary_col, and replaces with two \midrule
- doesn't appear to affect regular regression output as aligns are different

natashawatkins commented:
My output now from `summary_col.as_latex()`

![image](https://cloud.githubusercontent.com/assets/21038129/26478146/8ef0e332-420d-11e7-9d30-e33533fcd2f3.png)

josef-pkt commented:
Hi, I had only a brief look
style/pep-8: needs line break to stay below 80 characters per line

> doesn't appear to affect regular regression output as aligns are different

That sounds a bit fragile given that different summarys go through the same code. 
Is it possible to add a flag as attribute to `self` and condition on it, `if self.merge_latex_table: ...`?

Also can you make a simple unit test with just a few columns. Given that there doesn't seem to be anything that changes in the table, like dates or floating point noise, it would be easy just to compare the generated string. The regular expression might easily break if there are any formatting changes, and without unit tests it would be slow to be found.   
(Most of the time we only use smoke tests for summary, but in this case a full test to verify the output wouldn't be difficult.)

josef-pkt commented:
(insert comment before first comment :)

Thanks, the output looks much better after the change.

natashawatkins commented:
Thanks for your comments @josef-pkt 

I've gone ahead and added the changes you've suggested

The unit tests simply test whether the \end{tabular}\begin{tabular} still exist or not for OLS vs. summary_col output.

I've also deleted brackets causing the caption error (#3700) because it keeps giving me compilation issues and it's a very simple fix.

josef-pkt commented on statsmodels/iolib/tests/test_summary2.py:
```diff
@@ -0,0 +1,52 @@
+import unittest
+from statsmodels.regression.linear_model import OLS, add_constant
+from summary2 import summary_col
```
needs full import path, fails on all travis runs

natashawatkins replied:
oops! done thanks

josef-pkt commented:
looks good to me, one import error that needs to be fixed (travis test errors)

can you squash this (so we don't have the reverted file copy and redundant changes)

josef-pkt commented:
@natashawatkins  I merged this after rebase and some changes to the unit tests in #3963

Thank you for the contribution
